### DISCUSSION TITLE
Power Button

### DISCUSSION THREAD

**Discussion by rern** [rern/rAudio#181] [Tips]
**Power button** without extra circuit/components

_System > GPIO Devices > Power Button:_
		
![power_button-led](https://github.com/rern/_assets/raw/master/power_button-led.svg)

- Momentary switch
	- ON: Connect between fixed pin `5` and ground.
	- OFF: Connect between selectable pin and ground.
	- DPST / DPDT, 2 contacts switch, combines both into 1 button.
- LED (optional)
	- Connect between selectable pin and ground.
		- On when boot finished.
		- 12V integrated LED push button can be used.
		- Caution: a current limiting resister needed.

**Comment by Bart-Smith** [comment]
Connecting to PIN 5 ( GPIO3 / SCL ) does not work in combination of a DAC HAT.

**Comment by GlennFR** [comment]
I have found a workaround -

- The power-on function seems to be part of the RPi hardware and doesn't need to be declared.  Don't add any other hardware to GPIO3 (pin5) other than your pushbutton or intermittent switch to ground.  This pin is also one of the I2C pins used by many HATs.
- For the power-down function, you first need to find an unused pin!  Many DAC HATs have other built-in functions, like rotary controls for volume, IR sensors, etc...  On my IqAudio DAC I found that GPIO5 (pin29) isn't used.
- Instead of declaring in the RAudio interface, I added the following declaration in BOOT/config.txt BEFORE the line that activates the DAC HAT

`# See if we can get power-down to work.  GPIO5=header pin29`
`dtoverlay=gpio-shutdown,gpio_pin=5`
`dtoverlay=iqaudio-dac # < This is my DAC declaration line`

The DAC now works properly, and an intermittent connection pin29 to GND shuts the system down in a couple of seconds.  If power is still applied, an intermittent connection pin5 to GND restarts the system.

**Reply by rern** [reply]
Thank you for sharing.
Inserting `dtoverlay=gpio-shutdown,gpio_pin=5` will be included with next update.

**Reply by GlennFR** [reply]
Perhaps you need to add a comment to say that "gpio_pin=5" may have to be changed if a user has a configuration that already uses GPIO5/pin29 for something else...
Thx
